packages/main/src/Dialog.ts

```diff
@@ -340,7 +342,10 @@ class Dialog extends Popup {
 	}
 
 	get _role() {
-		return (this.state === ValueState.Error || this.state === ValueState.Warning) ? "alertdialog" : "dialog";
+		if (this.accessibleRole === PopupAccessibleRole.None) {
+			return undefined;
+		}
+		return (this.state === ValueState.Error || this.state === ValueState.Warning) ? PopupAccessibleRole.AlertDialog.toLowerCase() : this.accessibleRole.toLowerCase();
```

Comment:
it would be easier to read if you write the state check in a separate if statement